bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/factory/CaddxHandlerFactory.java

```diff
@@ -0,0 +1,261 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.caddx.internal.factory;
+
+import static org.openhab.binding.caddx.internal.CaddxBindingConstants.SUPPORTED_THING_TYPES_UIDS;
+
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.eclipse.smarthome.io.transport.serial.SerialPortManager;
+import org.openhab.binding.caddx.internal.CaddxBindingConstants;
+import org.openhab.binding.caddx.internal.config.CaddxBridgeConfiguration;
+import org.openhab.binding.caddx.internal.config.CaddxPartitionConfiguration;
+import org.openhab.binding.caddx.internal.config.CaddxZoneConfiguration;
+import org.openhab.binding.caddx.internal.discovery.CaddxDiscoveryService;
+import org.openhab.binding.caddx.internal.handler.CaddxBridgeHandler;
+import org.openhab.binding.caddx.internal.handler.ThingHandlerKeypad;
+import org.openhab.binding.caddx.internal.handler.ThingHandlerPanel;
+import org.openhab.binding.caddx.internal.handler.ThingHandlerPartition;
+import org.openhab.binding.caddx.internal.handler.ThingHandlerZone;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link CaddxHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Georgios Moutsos - Initial contribution
+ */
+@Component(configurationPid = "binding.caddx", service = ThingHandlerFactory.class)
+@NonNullByDefault
+public class CaddxHandlerFactory extends BaseThingHandlerFactory {
+    private final Logger logger = LoggerFactory.getLogger(CaddxHandlerFactory.class);
+    private final Map<ThingUID, ServiceRegistration<?>> discoveryServiceRegistrations = new HashMap<>();
+    private @NonNullByDefault({}) SerialPortManager portManager;
+
+    @Override
+    public @Nullable Thing createThing(ThingTypeUID thingTypeUID, Configuration configuration,
+            @Nullable ThingUID thingUID, @Nullable ThingUID bridgeUID) {
```

Comment:
Why are you overriding the bridge ids that are passed to this method?

Reply:
That's also code i copied from the dscalarm binding.
I suppose there is no need to override the createThing function at all. I'm inclined to delete this function. Is there a reason not to?

Reply:
I think it should deleted as well.

Reply:
Deleted